File: generated-components/rsc/Button.js
"use client";
import * as React from "react";
import { useState } from "react";

function Button(props: any) {
  const [isPressed, setIsPressed] = useState(() => false);

  function handleMouseDown() {
    setIsPressed(true);
  }

  function handleMouseUp() {
    setIsPressed(false);
  }

  return (
    <button
      style={getButtonStyle()}
      onClick={(event) => (props.disabled ? undefined : props.onClick)()}
      onMouseDown={(event) => handleMouseDown()}
      onMouseUp={(event) => handleMouseUp()}
      disabled={props.disabled}
    >
      {props.loading ? (
        <span>Loading...</span>
      ) : (
        <>{props.children || props.text}</>
      )}
    </button>
  );
}

export default Button;

File: generated-components/swift/Button.swift
import SwiftUI

struct Button: View {
  let disabled: Any
  let onClick: Any
  let loading: Any
  let children: Any
  let text: Any

  @State private var isPressed: Bool = false

  // Function state variables
  var handleMouseDown: () -> Void = { () -> Void in
    self.isPressed = true;
  }
  var handleMouseUp: () -> Void = { () -> Void in
    self.isPressed = false;
  }

  var body: some View {
    Button(action: { (self.disabled ? undefined : self.onClick()) }) {
      Show {
        Text("")
      }
    }
    // Dynamic styles not fully implemented
    .modifier(/* Dynamic style handling needed here */)"
    .onMouseDown(self.handleMouseDown())
    .onMouseUp(self.handleMouseUp())

    }

    }

    #if DEBUG
    struct Button_Previews: PreviewProvider {
      static var previews: some View {
        Button(
          disabled: /* provide preview value */,
          onClick: /* provide preview value */,
          loading: /* provide preview value */,
          children: /* provide preview value */,
          text: /* provide preview value */
        )
      }
    }
    #endif